### PR TITLE
[chore] [receiver/simpleprometheus] Use confighttp.NewDefaultClientConfig instead of manually creating struct

### PR DESCRIPTION
**Pull request by mackjmr**
**Description:**
This PR makes usage of `NewDefaultClientConfig` instead of manually creating the confighttp.ClientConfig struct.

**Link to tracking Issue:** #35457

### CODE REVIEW

**Comment by mackjmr** [comment]
@fatsheep9146 can you please take a look ?

**Review by fatsheep9146 — COMMENTED** [review]
I think integration test is broken, could you please check this?
> === FAIL: . TestGetPrometheusConfigWrapper/Test_TLSEnable_false_but_tls_configured (re-run 1) (0.00s)
    receiver_test.go:437: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/simpleprometheusreceiver/receiver_test.go:437
        	Error:      	Not equal: 
        	            	expected: &prometheusreceiver.Config{PrometheusConfig:(*prometheusreceiver.PromConfig)(0xc00057e788), TrimMetricSuffixes:false, UseStartTimeMetric:false, StartTimeMetricRegex:"", ReportExtraScrapeMetrics:false, TargetAllocator:(*targetallocator.Config)(nil)}
        	            	actual  : &prometheusreceiver.Config{PrometheusConfig:(*prometheusreceiver.PromConfig)(0xc00057ec88), TrimMetricSuffixes:false, UseStartTimeMetric:false, StartTimeMetricRegex:"", ReportExtraScrapeMetrics:false, TargetAllocator:(*targetallocator.Config)(nil)}
        	                 	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -48,3 +48,3 @@
        	            	     MetricsPath: (string) (len=7) "/metric",
        	            	-    Scheme: (string) (len=5) "https",
        	            	+    Scheme: (string) (len=4) "http",
        	            	     EnableCompression: (bool) false,

**Comment by mackjmr** [comment]
Thanks @fatsheep9146, fixed in latest commit 👍

File: receiver/simpleprometheusreceiver/receiver_test.go [hunk 1]
```diff
@@ -348,12 +360,7 @@ func TestGetPrometheusConfigWrapper(t *testing.T) {
 				httpConfig: httpConfig{
 					TLSEnabled: false,
 				},
-				ClientConfig: confighttp.ClientConfig{
-					Endpoint: defaultEndpoint,
-					TLSSetting: configtls.ClientConfig{
-						Insecure: false,
-					},
-				},
+				ClientConfig:       clientConfig,
```
Comment (mackjmr) [inline receiver/simpleprometheusreceiver/receiver_test.go:363]: Thanks @fatsheep9146, fixed here 👍

**Review by fatsheep9146 — APPROVED** [review]

**Comment by fatsheep9146** [comment]
@open-telemetry/collector-contrib-maintainers Could any maintainers help merge this pr please? Thanks!

**Review by mx-psi — APPROVED** [review]

**Pull request merged**